test/test_bandedlu.jl

```diff
@@ -24,11 +24,7 @@ let
     @test BandedMatrices._promote_to_blas_type(typ, Int16)      == typ
     @test BandedMatrices._promote_to_blas_type(typ, Int8)       == typ
 
-    if VERSION < v"0.5.0-dev+5122"
```

Comment:
The issue is that eventually the package itself may no longer work in a given version, say when the package goes 0.6 only.  Once that happens, having special cases in the tests for versions of Julia where the package doesn't even work no longer makes sense.

Maybe 0.6 final is a reasonable time frame for dropping the special case?